integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java

```diff
@@ -0,0 +1,502 @@
+        () -> exec(new String(deployTlsIngress), true));
+
+    logger.info("kubectl apply returned {0}", result.toString());
+    checkServiceExists("voyager-voyager-tls-tunneling", domainNamespace);
+
+    int httpsTunnelingPort =
+        getServiceNodePort(domainNamespace, "voyager-voyager-tls-tunneling");
+    assertTrue(httpsTunnelingPort != -1,
+        "Could not get the HttpsTunnelingPort service node port");
+    logger.info("HttpsTunnelingPort for Voyager {0}", httpsTunnelingPort);
+
+    StringBuffer httpsUrl = new StringBuffer("https://");
+    httpsUrl.append(K8S_NODEPORT_HOST + ":" + httpsTunnelingPort);
+
+    StringBuffer javasCmd = new StringBuffer("java -cp ");
+    javasCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar"));
+    javasCmd.append(":");
+    javasCmd.append(Paths.get(RESULTS_ROOT));
+    // javasCmd.append(" -Djavax.net.debug=all ");
+    javasCmd.append(" -Djavax.net.ssl.trustStorePassword=password");
+    javasCmd.append(" -Djavax.net.ssl.trustStoreType=jks");
+    javasCmd.append(" -Djavax.net.ssl.trustStore=");
+    javasCmd.append(jksTrustFile);
+    javasCmd.append(" JmsTestClient ");
+    javasCmd.append(httpsUrl);
+    logger.info("java command to be run {0}", javasCmd.toString());
+
+    // Note it takes a couples of iterations before the client success
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Wait for Https JMS Client to access WLS "
+                    + "(elapsed time {0}ms, remaining time {1}ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(runJmsClient(new String(javasCmd)));
+  }
+
+  // Download the wlthint3client.jar from Adminserver pod to local filesystem.
+  // Use wlthint3client.jar in classpath to build and run the standalone 
+  // JMS client that sends messages to a Uniform Distributed Queue using 
+  // load balancer http(s) url which maps to custom channel on cluster member
+  // server on WebLogic cluster.
+  private void buildClient() {
+
+    assertDoesNotThrow(() -> copyFileFromPod(domainNamespace, 
+             adminServerPodName, "weblogic-server",
+             "/u01/oracle/wlserver/server/lib/wlthint3client.jar", 
+             Paths.get(RESULTS_ROOT, "wlthint3client.jar")));
+    StringBuffer javacCmd = new StringBuffer("javac -cp ");
+    javacCmd.append(Paths.get(RESULTS_ROOT, "wlthint3client.jar "));
+    javacCmd.append(Paths.get(RESOURCE_DIR, "tunneling", "JmsTestClient.java"));
+    javacCmd.append(Paths.get(" -d "));
+    javacCmd.append(Paths.get(RESULTS_ROOT));
+    logger.info("javac command {0}", javacCmd.toString());
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(javacCmd), true));
+    logger.info("javac returned {0}", result.toString());
+    logger.info("javac returned EXIT value {0}", result.exitValue());
+    assertTrue(result.exitValue() == 0, "Client compilation fails");
```

Comment:
log the error that it fails with

Reply:
Modified the logic to put stderr and stdout from ExecResult class

Reply:
Modified logic to use actions.impl.primitive.Command to execute bash command.